provider/openstack/provider.go

```diff
@@ -1147,23 +1147,31 @@ func (e *Environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 			if err != nil {
 				break
 			}
+			if server == nil {
+				continue
```

Comment:
when testing with instrumented code, this causes juju to not delete the instance that was created.  suggest a message to the user, to look for stray instances.

i believe the break should be used, it will allow the provisioner to retry, though an instance will still be left behind. Given the past changes to parallel the provisioner, perhaps the for loop here should have been removed then, as retries should be handled by the it. maybe a change for another day.

Reply:
I agree - I'll add a WARN log message to look for stray instances and will change 'continue' to 'break'. 

As for the rest of the comment, is it possible for you to file a tech debt bug describing the improvement and its benefit?